src/libclipper/include/clipper/threadpool.hpp

```diff
@@ -210,41 +213,78 @@ class ThreadPool {
     PackagedTask task{std::move(boundTask)};
     auto result_future = task.get_future();
 
-    work_queue_.push(std::make_unique<TaskType>(std::move(task)));
+    size_t queue_id = get_queue_id(vm, replica_id);
+    boost::shared_lock<boost::shared_mutex> l(queues_mutex_);
+    auto queue = queues_.find(queue_id);
+    if (queue != queues_.end()) {
+      queue->second.push(std::make_unique<TaskType>(std::move(task)));
+    } else {
+      std::stringstream error_msg;
+      error_msg << "No work queue for model " << versioned_model_to_str(vm)
+                << ", replica " << std::to_string(replica_id);
+      log_error(LOGGING_TAG_THREADPOOL, error_msg.str());
+      throw std::runtime_error(error_msg.str());
+    }
+    // work_queue_.push(std::make_unique<TaskType>(std::move(task)));
     return result_future;
   }
 
+  static size_t get_queue_id(const VersionedModelId& vm, const int replica_id) {
```

Comment:
Is this hash effective? I remember encountering xor-based hash collision issues when hashing our input vectors that resulted in caching problems. Consider using `boost::hash_combine` instead.